doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+<https://packaging.python.org/tutorials/packaging-projects/>`_, and create some empty
+files for us to fill in, like this::
+
+    my_package/
+        myapp/
+            __init__.py
+            __main__.py
+            app.py
+            plugin.py
+            fluxSolver.py
+            thermalSolver.py
+        setup.py
+        README.md
+        LICENSE.md
+
+
+These files are:
+
+* The outer :file:`my_package` root directory is a container for your app. The name does not
+  matter to ARMI; you can rename it to anything.
+
+* The inner :file:`myapp` directory is the actual Python package for your app. Its name is the
+  Python package name you will use to import anything inside (e.g. ``myapp.plugins``).
+
+* :file:`myapp/__init__.py` tells Python that this directory is a Python package and registers
+  the application with the ARMI framework.
+
+* :file:`myapp/__main__.py` provides one or more entry points for users of your app (including
+  you!) to start running it.
+
+* :file:`myapp/app.py` contains the actual app registration code that will be called by
+  :file:`__init__.py`. This can be named anything as long as it is consistent with the
+  registration code in :file:`myapp/__init__.py`. 
+
+* :file:`myapp/plugin.py` contains the code that defines the physics plugin we will create
+
+* :file:`myapp/fluxSolver.py` contains the flux solver
+
+* :file:`myapp/thermalSolver.py` contains the thermal/hydraulics solver
+
+* :file:`setup.py` the python package installation file to help users install your
+  application.
+
+* :file:`README.md` and :file:`LICENSE.md` are an optional description and license of your
+  application that would be prominently featured, e.g. in a github repo, if you were to
+  put it there.
+
+Registering the app with ARMI
+=============================
+The ARMI Framework contains features to run the "main loop" of typical applications. In
+order to get access to these, we must register our new app with the ARMI framework. To do
+this, we put the following code in the top-level :file:`__init__.py` module:
+
+.. code-block:: python
+    :caption: ``myapp/__init__.py``
+
+    import armi
+    from myapp import app
+    armi.configure(app.ExampleApp())
+
+Defining the app class
+======================
+We define our app in the :file:`myapp/app.py` module. For this example, the app class is
+relatively simple: it will just register our one custom plugin. We will actually create
+the plugin shortly.
+
+.. admonition:: Apps vs. plugins vs. interfaces
+
+    ARMI-based methodologies are broken down into three layers of abstraction. Apps are
+    collections of plugins intended to perform analysis on a certain type of reactor.
+    Plugins are independent and mixable collections of relatively arbitrary code that
+    might bring in special materials, contain certain engineering methodologies, and/or
+    Interfaces with one or more physics kernels.
```

Comment:
To add to this: 

A plugin can typically contain one or more interfaces that behave similarly (i.e., neutronics plugin) and an app typically contains one or more plugins. As mentioned above, this example application will implement a `DummyPhysicsPlugin`, which will implement **both** a neutronics and thermal-hydraulics interface.

Reply:
I feel like that's too much of a wall of text for a first intro but will link to the framework docs for more info. I also am hesitant to say "typically" because other orgs may do things way differently than us.